### PR TITLE
fix: resolve bug where changing duration when filter string is applied does not change data (original duration is still applied)

### PR DESCRIPTION
**Pull request by lynnshaoyu**
https://uniswaplabs.atlassian.net/browse/WEB-1371?atlOrigin=eyJpIjoiZGRhNjMxMmMyZGE4NGI4MTgyOGQyMTgwODI4ZTUxNjAiLCJwIjoiaiJ9

### CODE REVIEW

File: src/graphql/data/TopTokens.ts [hunk 1]
```diff
@@ -132,18 +132,18 @@ function toContractInput(token: PrefetchedTopToken) {
...
 }
 
-// Map of key: ${chain} + ${address} and value: TopToken object.
+// Map of key: ${HistoryDuration} and value: another Map, of key:${chain} + ${address} and value: TopToken object.
 // Acts as a local cache.
-const tokensWithPriceHistoryCache: Record<string, TopToken> = {}
+const tokensWithPriceHistoryCache: Record<string, Record<string, TopToken>> = {}
 
-const checkIfAllTokensCached = (tokens: PrefetchedTopToken[]) => {
+const checkIfAllTokensCached = (duration: string, tokens: PrefetchedTopToken[]) => {
```
Comment (JFrankfurt) [inline src/graphql/data/TopTokens.ts:139]: iirc there is a HistoryDuration type somewhere that we could use here

Comment (lynnshaoyu) [reply src/graphql/data/TopTokens.ts:139]: yeah although typescript  forces us to predefine the key values and include '%future added value'. I guess this works fine though? (changed it)

File: src/graphql/data/TopTokens.ts [hunk 2]
```diff
@@ -132,18 +132,27 @@ function toContractInput(token: PrefetchedTopToken) {
...
+
+const checkIfAllTokensCached = (duration: HistoryDuration, tokens: PrefetchedTopToken[]) => {
   let everyTokenInCache = true
   const cachedTokens: TopToken[] = []
 
   const checkCache = (token: PrefetchedTopToken) => {
     const tokenCacheKey = !!token ? `${token.chain}${token.address}` : ''
-    if (tokenCacheKey in tokensWithPriceHistoryCache) {
-      cachedTokens.push(tokensWithPriceHistoryCache[tokenCacheKey])
+    if (duration in tokensWithPriceHistoryCache && tokenCacheKey in tokensWithPriceHistoryCache[duration]) {
```
Comment (JFrankfurt) [inline src/graphql/data/TopTokens.ts:154]: will `duration in tokensWithPriceHistoryCache` always return true now that you've got a default value?

Comment (lynnshaoyu) [reply src/graphql/data/TopTokens.ts:154]: yes true, fixed here and line 215. ty for catching.

**Review by cmcewen — APPROVED** [review]

**Pull request merged**